### PR TITLE
docs: update configuration part

### PR DESCRIPTION
**Pull request by promer94**

### CODE REVIEW

File: packages/nextra-theme-docs/src/types.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
+import React from 'react'
+
 export interface DocsThemeConfig {
+  projectLink?: string
+  github?: string
+  projectLinkIcon?: React.ReactNode | React.FC<{ locale?: string }>
   docsRepositoryBase?: string
   titleSuffix?:
-    | string
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/types.ts:4]: why this change? according to types now specifying `titleSuffix` as string will be incorrect

File: examples/docs/src/pages/features/mdx.mdx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+Renders
+
+- [x] Write the press release
+- [ ] Update the website
+- [ ] Contact the media
+
+## Syntax Highlighting
+
+Automatica syntax highlighting:
```
Comment (B2o5T) [inline examples/docs/src/pages/features/mdx.mdx:78]:
```suggestion
Automatic syntax highlighting:
```

File: examples/docs/src/pages/features/mdx.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+```
+````
+
+Renders:
+
+```js
+console.log('hello, world')
+```
+
+You can also add the `highlight=<line|range>` modifier to highlight specific lines:
```
Comment (B2o5T) [inline examples/docs/src/pages/features/mdx.mdx:92]:
```suggestion
You can also add the `{line|range}` modifier to highlight specific lines:
```

File: examples/docs/src/pages/get-started.mdx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+1. Install Next.js, Nextra and React
+
+<Nextra.Tabs items={['npm', "yarn", "pnpm"]} defaultIndex={1}>
+  <Nextra.Tab>
+    ```bash
+    npm i react react-dom next nextra
+    ```
+  </Nextra.Tab>
+  <Nextra.Tab>
+     ```bash
```
Comment (B2o5T) [inline examples/docs/src/pages/get-started.mdx:25]:
```suggestion
    ```bash
```

File: examples/docs/src/pages/get-started.mdx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+2. Install the docs theme
+
+<Nextra.Tabs items={['npm', "yarn", "pnpm"]} defaultIndex={1}>
+  <Nextra.Tab>
+    ```bash
+    npm i nextra-theme-docs
+    ```
+  </Nextra.Tab>
+  <Nextra.Tab>
+     ```bash
```
Comment (B2o5T) [inline examples/docs/src/pages/get-started.mdx:46]:
```suggestion
    ```bash
```

File: examples/docs/src/pages/themes/docs/bleed.mdx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  ![Landscape](https://source.unsplash.com/eaxwP9J_V6s/1600x398)
+</Nextra.Bleed>
+
+<Nextra.Bleed full>
+  <iframe
+    src="https://codesandbox.io/embed/swr-states-4une7"
+    width="100%"
+    height="500px"
+    title="SWR-States"
+  ></iframe>
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/bleed.mdx:59]:
```suggestion
  />
```

File: examples/docs/src/pages/themes/docs/bleed.mdx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  ![Landscape](https://source.unsplash.com/eaxwP9J_V6s/1600x398)
+</Bleed>
+
+<Bleed full>
+  <iframe
+    src="https://codesandbox.io/embed/swr-states-4une7"
+    width="100%"
+    height="500px"
+    title="SWR-States"
+  ></iframe>
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/bleed.mdx:80]:
```suggestion
  />
```

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 8]
```diff
@@ -0,0 +1,618 @@
...
+**Example:**
+
+```jsx filename="theme.config.js"
+/**
+ * @type {import('nextra-theme-docs').DocsThemeConfig}
+ */
+export default {
+  font: true
+}
+```
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:258]: currently, there is no logic that handles `fonts`, so this part can be commented

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 9]
```diff
@@ -0,0 +1,618 @@
...
+
+## `feedbackLink`
+
+The components that should be shown on the link that leads to the issues or discussions of the repository.
+
+**Type:** `ReactNode | React.FC<{ locale?: string }>{:ts}`
+
+**Default:** `Edit this page`
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:321]:
```suggestion
**Default:** `Feedback`
```
https://github.com/shuding/nextra/blob/core/packages/nextra-theme-docs/src/toc.tsx#L113

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 10]
```diff
@@ -0,0 +1,618 @@
...
+}
+```
+
+## `feedbackLabels`
+
+Label used for create issue or discussion.
+
+**Type:** `string`
+
+**Default:** `feedback`
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:340]:
```suggestion
**Default:** ``
```
https://github.com/shuding/nextra/blob/core/packages/nextra-theme-docs/src/toc.tsx#L52

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 11]
```diff
@@ -0,0 +1,618 @@
...
+ * @type {import('nextra-theme-docs').DocsThemeConfig}
+ */
+export default {
+  logo: ({ locale }) => (
+    <React.Fragment>
+      <span className="mr-2 font-extrabold hidden md:inline">Nextra</span>
+      <span className="text-gray-600 font-normal hidden md:inline">
+        The Next Docs Builder
+      </span>
+    </React.Fragment>
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:372]:
```suggestion
    <>
      <span className="mr-2 font-extrabold hidden md:inline">Nextra</span>
      <span className="text-gray-600 font-normal hidden md:inline">
        The Next Docs Builder
      </span>
    </>
```

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 12]
```diff
@@ -0,0 +1,618 @@
...
+      <meta name="theme-color" content="#ffffff" />
+      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+      <meta httpEquiv="Content-Language" content="en" />
+      <meta name="description" content="Nextra: the next docs builder" />
+      <meta name="twitter:card" content="summary_large_image" />
+      <meta name="twitter:site" content="@shuding_" />
+      <meta property="og:title" content="Nextra: the next docs builder" />
+      <meta property="og:description" content="Nextra: the next docs builder" />
+      <meta name="apple-mobile-web-app-title" content="Nextra" />
+    </React.Fragment>
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:402]:
```suggestion
    <>
      <meta name="msapplication-TileColor" content="#ffffff" />
      <meta name="theme-color" content="#ffffff" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta httpEquiv="Content-Language" content="en" />
      <meta name="description" content="Nextra: the next docs builder" />
      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:site" content="@shuding_" />
      <meta property="og:title" content="Nextra: the next docs builder" />
      <meta property="og:description" content="Nextra: the next docs builder" />
      <meta name="apple-mobile-web-app-title" content="Nextra" />
    </>
```

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 13]
```diff
@@ -0,0 +1,618 @@
...
+}
+```
+
+## `projectChatLink`
+
+The URL that the chat button in the top right will point to.
+
+**Type:** `string`
+
+**Default:** `https://discord.gg/hEM84NMkRv`
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:454]:
```suggestion
**Default:** ``
```
I guess it's not specified

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 14]
```diff
@@ -0,0 +1,618 @@
...
+  projectChatLinkIcon: <Discord />
+}
+```
+
+## `banner`
+
+The banner content that will display at top of the page.
+
+**Type:** `ReactNode | React.FC<{ locale?: string }>{:ts}`
+
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:489]:
```suggestion

**Default:** `'nextra-banner'`

```

File: examples/docs/src/pages/themes/docs/configuration.mdx [hunk 15]
```diff
@@ -0,0 +1,618 @@
...
+  banner: ({ locale }) => 'Nextra 2'
+}
+```
+
+## `bannerKey`
+
+The key that is used to control the display of banner in localStorage.
+
+**Type:** `string`
+
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/configuration.mdx:506]:
```suggestion

**Default:** `'nextra-banner'`

```

File: examples/docs/src/pages/themes/docs/index.mdx [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+1. Install Next.js, Nextra and React
+
+<Nextra.Tabs items={['npm', "yarn", "pnpm"]} defaultIndex={1}>
+  <Nextra.Tab>
+    ```bash
+    npm i react react-dom next nextra
+    ```
+  </Nextra.Tab>
+  <Nextra.Tab>
+     ```bash
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/index.mdx:18]:
```suggestion
    ```bash
```

File: examples/docs/src/pages/themes/docs/index.mdx [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+2. Install the docs theme
+
+<Nextra.Tabs items={['npm', "yarn", "pnpm"]} defaultIndex={1}>
+  <Nextra.Tab>
+    ```bash
+    npm i nextra-theme-docs
+    ```
+  </Nextra.Tab>
+  <Nextra.Tab>
+     ```bash
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/index.mdx:39]:
```suggestion
    ```bash
```

File: examples/docs/src/pages/themes/docs/meta.json [hunk 18]
```diff
@@ -0,0 +1,8 @@
+{
+  "index": "Installation",
+  "configuration": "Configuration",
+  "callout": "Callout",
+  "bleed": {
+    "title": "Bleed"
+  }
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/meta.json:7]:
```suggestion
  "bleed": "Bleed"
```

File: examples/docs/src/pages/themes/docs/tabs.mdx [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+    ```c filename="hi.c"
+    #include <stdio.h>
+    ```
+  </Nextra.Tab>
+  <Nextra.Tab>
+    ```python filename="hello.py"
+    print('Hello, world!')
+    ```
+  </Nextra.Tab>
+````
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/tabs.mdx:55]:
`````suggestion
  </Nextra.Tab>
</Nextra.Tabs>
````
`````

File: examples/docs/src/pages/themes/docs/tabs.mdx [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+import Tabs from 'nextra-docs-theme/Tabs'
+const items = ['1', '2', '3', '4']
+const defaultIndex = 1
+const Component = () => (
+  <Tabs items={items} defaultIndex={defaultIndex}>
+    <Tabs.Tab>1</Tab>
+    <Tabs.Tab>2</Tab>
+    <Tabs.Tab>3</Tab>
+    <Tabs.Tab>4</Tab>
+  </Tabs>
```
Comment (B2o5T) [inline examples/docs/src/pages/themes/docs/tabs.mdx:69]:
```suggestion
import { Tabs, Tab } from 'nextra-docs-theme/Tabs'
const items = ['1', '2', '3', '4']
const defaultIndex = 1
const Component = () => (
  <Tabs items={items} defaultIndex={defaultIndex}>
    <Tab>1</Tab>
    <Tab>2</Tab>
    <Tab>3</Tab>
    <Tab>4</Tab>
  </Tabs>
```
there is no `export default` in `tabs.tsx`

File: examples/docs/src/theme.config.js [hunk 21]
```diff
@@ -32,7 +67,9 @@ export default {
   nextLinks: true,
   floatTOC: true,
   footer: true,
-  footerEditLink: ({ locale }) =>
-    locale === 'zh' ? '前往 GitHub 编辑此页' : 'Edit this page on GitHub',
-  footerText: <>MIT {new Date().getFullYear()} © Shu Ding.</>
+  footerEditLink: 'Edit this page on GitHub',
+  footerText: <>MIT {new Date().getFullYear()} © Nextra.</>,
```
Comment (B2o5T) [inline examples/docs/src/theme.config.js:71]:
```suggestion
  footerText: `MIT ${new Date().getFullYear()} © Nextra.`,
```

File: packages/nextra-theme-docs/src/utils/render-component.tsx [hunk 22]
```diff
@@ -1,8 +1,8 @@
-import React from 'react'
+import React, { PropsWithChildren } from 'react'
 
 const renderComponent = <T,>(
-  ComponentOrNode: React.FC<T> | React.ReactNode,
-  props: T,
+  ComponentOrNode: React.FC<PropsWithChildren<T>> | React.ReactNode,
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/utils/render-component.tsx:4]:
```suggestion
  ComponentOrNode: React.FC<T> | React.ReactNode,
```
image

File: packages/nextra/src/loader.ts [hunk 23]
```diff
@@ -186,6 +186,7 @@ const NextraLayout = __nextra_withSSG__(__nextra_withLayout__({
   headings: ${JSON.stringify(headings)},
   hasH1: ${JSON.stringify(hasH1)},
   ${timestamp ? `timestamp: ${timestamp},\n` : ''}
+  ${unstable_flexsearch ? `unstable_flexsearch: ${JSON.stringify(unstable_flexsearch)}` : ''}
```
Comment (B2o5T) [inline packages/nextra/src/loader.ts:189]: we don't need it since we specify also in `theme.config`, otherwise, this value should be removed from `theme.config` and we should keep only value from nextra config https://github.com/shuding/nextra/blob/core/packages/nextra-theme-docs/src/types.ts#L54

File: packages/nextra-theme-docs/src/utils/render-component.tsx (hunk 22 above)
Comment (promer94) [reply packages/nextra-theme-docs/src/utils/render-component.tsx:4]: It's for future react 18 update. I will update other FC types later

File: packages/nextra/src/loader.ts (hunk 23 above)
Comment (promer94) [reply packages/nextra/src/loader.ts:189]: We need to pass the option here. Otherwise user will need to configure twice which is confusing.

Comment (B2o5T) [reply packages/nextra/src/loader.ts:189]: @promer94 yes it's confusing, so let's remove this option from `theme.config`

**Comment by B2o5T** [comment]
also let's run prettier in another PR since it will make review difficult, also you should very carefully prettify MDX files with prettier since he doesn't support MDX2 and broke some stuff regularly

**Comment by promer94** [comment]
> also you should very carefully prettify MDX files with prettier since he doesn't support MDX2 and broke some stuff regularly

oh i didn't know that. Thanks for the tips ! 

I would like to merge it now and i will fix those format problem later. ❤️
